.github/workflows/plugin_upstream_testing_base.yml

```diff
@@ -45,78 +66,67 @@ jobs:
           builder: "${{ steps.buildx.outputs.name }}"
           context: "./"
           push: false
-          tags: "${{ inputs.plugin_name }}/nautobot:${{ matrix.nautobot-version }}-py${{ env.PYTHON_VERSION }}"
+          tags: "${{ env.APP_NAME }}/nautobot:${{ inputs.nautobot_branch }}-py${{ env.PYTHON_VERSION }}"
           file: "./development/Dockerfile"
-          cache-from: "type=gha,scope=nautobot-${{ matrix.nautobot-version }}"
+          cache-from: "type=gha,scope=nautobot-${{ inputs.nautobot_branch }}"
           build-args: |
-            NAUTOBOT_VER=${{ matrix.nautobot-version }}
+            NAUTOBOT_VER=${{ inputs.nautobot_branch }}
             PYTHON_VER=${{ env.PYTHON_VERSION }}
             CI=true
-          outputs: "type=docker,dest=/tmp/nautobot_${{ matrix.nautobot-version }}.tar"
+          outputs: "type=docker,dest=/tmp/nautobot_${{ inputs.nautobot_branch }}.tar"
       - name: Upload docker image artifact
         uses: actions/upload-artifact@v4
         with:
-          name: "docker_nautobot_${{ matrix.nautobot-version }}"
-          path: "/tmp/nautobot_${{ matrix.nautobot-version }}.tar"
+          name: "docker_nautobot_${{ inputs.nautobot_branch }}"
+          path: "/tmp/nautobot_${{ inputs.nautobot_branch }}.tar"
+          retention-days: 1 # Default is 90 days; we only need this for the following jobs in this workflow
   unittest:
     name: "Run Unit Tests"
     needs: "build"
-    strategy:
-      fail-fast: false
-      matrix:
-        nautobot-version: ["develop", "next"]
-    runs-on: "ubuntu-24.04"
+    runs-on: "ubuntu-latest"
     steps:
       # Try to download the artifact first so the job can be skipped if the artifact is not found
       - name: Download docker image artifact
         uses: actions/download-artifact@v4
         with:
-          name: "docker_nautobot_${{ matrix.nautobot-version }}"
+          name: "docker_nautobot_${{ inputs.nautobot_branch }}"
           path: /tmp
       - name: Load docker image
-        run: "docker load --input /tmp/nautobot_${{ matrix.nautobot-version }}.tar"
+        run: "docker load --input /tmp/nautobot_${{ inputs.nautobot_branch }}.tar"
       - name: "Check out repository code"
         uses: "actions/checkout@v4"
         with:
-          ref: "${{ matrix.nautobot-version == 'develop' && 'develop' || matrix.nautobot-version == 'next' && 'next-3.0' || env.GITHUB_REF_NAME || 'develop' }}"
+          ref: "${{ inputs.app_branch }}"
       - name: "Setup environment"
         uses: "networktocode/gh-action-setup-poetry-environment@v6"
         with:
           poetry-install-options: ""  # override default "--only dev"
           poetry-version: "2.1.4"
           python-version: "${{ env.PYTHON_VERSION }}"
-      - name: "Copy credentials"
-        run: "cp development/creds.example.env development/creds.env"
```

Comment:
Why is this no longer needed?

Reply:
Done automatically via invoke (`build` I think) nowadays.